platform/engineering/request-for-change/2019-09-30-cms-integration.md

```diff
@@ -0,0 +1,230 @@
+`vets-website` pulls the content to serve up, it's already been built
+and validated. **Build failures due to content issues will be isolated
+to the content build and will not affect `vets-website` builds or
+deploys.** This will improve the overall stability of the
+`vets-website` CI.
+
+## Background
+The current `vets-website` build script looks like this.
+
+![current build](images/vets-website-build-before-separation.png)
+
+### Glossary
+This RFC will use the following terms. These terms may or may not be
+used outside of this RFC.
+- Full Jenkins build
+  - All branches are build using this pipeline (including `master`)
+  - This attempts to pull all CMS content, but will fall back to the
+    cache if necessary
+  - This builds JavaScript and CSS assets
+    - For the build of the `master` branch, these assets are saved in
+      S3
+  - Runs the full suite of unit, e2e, and accessibility tests
+- Content-only build
+  - Used in a content-only deploy
+  - Attempts to pull the latest CMS content
+    - Will not fall back to the cache
+  - Uses the JavaScript and CSS assets from the latest full build
+  - Runs a limited suite of tests
+- React app landing page
+  - An HTML file used as the landing page for a React application
+  - Has the normal header and footer
+  - Points to the appropriate assets for the application
+
+### Metalsmith
+Currently, `vets-website`'s build is managed by Metalsmith, a
+plugin-based static site generator. Metalsmith uses a number of custom
+plugins used to manipulate the static content in various ways. One
+such plugin fetches the content from Drupal if we pass the
+`--pull-drupal` flag to the build script (`yarn build --pull-drupal`)
+or if the local cache isn't found, another checks for broken links,
+etc.
+
+This build script is run in a number of ways:
+- The Jenkins pipeline for a full build
+- The Jenkins pipeline for a content-only build
+- Locally, with `yarn build` or `yarn watch`
+
+Separate, but related, the [preview
+server](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/platform/architecture/cms/#preview-server)
+has a modified version of the Metalsmith build (`yarn preview`). This
+is used for content editors to see how their changes show up on the
+site.
+
+Having a Metalsmith build script came from a time when `vets-website`
+had a `content/` directory from which the static pages were
+built. These days, however, all of the content comes from outside
+`vets-website`.
+
+### Content validation
+Some basic validations are performed on the content from the CMS and
+vagov-content repo.
+
+| Validation | Full build | Content-only build | Preview server |
+| ---------- | ---------- | ------------------ | -------------- |
+| Broken link checking | True; will fail the build on the `master` branch only | True; will fail the build | False |
+| Accessibility | True; will fail the build | True; will fail the build | False |
```

Comment:
@ncksllvn and @1Copenut are working on a way to add axe errors and warnings to the CMS preview server: https://github.com/department-of-veterans-affairs/vets-website/pull/10836

I think this inline error checking should go forward, regardless of combined/split build tasks.

Reply:
Agreed. That's a great addition to the preview server! As it stands now, though, the preview server doesn't do that checking, right?

Reply:
@cvalarida Correct, as it stands right now, the preview server does not do this checking.

Reply:
Cool. I'll go ahead and leave that out of this for now, then, since I want it to reflect the current state. This RFC shouldn't affect adding that functionality.